src/utils/netutils.cpp

```diff
@@ -157,56 +164,69 @@ namespace
     }
 } // namespace
 
-char const* resolve_ipv4_address(const char* ip, in_addr & s4_sin_addr)
+AddrInfoPtrWithDel_t
+resolve_ip_address(const char *ip, int port, bool is_ipv6_target_conn, const char **error_result) noexcept
 {
-    if (!inet_aton(ip, &s4_sin_addr)) {
-        struct addrinfo * addr_info = nullptr;
-        int               result    = getaddrinfo(ip, nullptr, nullptr, &addr_info);
-        if (result) {
-            char const* error = (result == EAI_SYSTEM) ? strerror(errno) : gai_strerror(result);
-            LOG(LOG_ERR, "DNS resolution failed for %s with errno = %d (%s)",
-                ip, (result == EAI_SYSTEM) ? errno : result, error);
-            return error;
+    AddrInfoPtrWithDel_t addr_info_ptr;
+    addrinfo *addr_info = nullptr;
+    addrinfo hints;
+    char port_buf[16] { };
+
+    std::memset(&hints, 0, sizeof(hints));
```

Comment:
To replace with `addrinfo hints {};`

Reply:
On C API man, it uses memset but yes, we are in C++ now !